File: LeetCode/Easy/MiddleOfLinkedList.php
/**
 * Definition for a singly-linked list.
 * class ListNode {
 *     public $val = 0;
 *     public $next = null;
 *     function __construct($val = 0, $next = null) {
 *         $this->val = $val;
 *         $this->next = $next;
 *     }
 * }
 */
class Solution
{

    /**
     * @param ListNode $head
     * @return ListNode
     */
    function middleNode( $head )
    {
        $count = 0;
        $curr = $head;
        
        while ($curr->next != null)
        {
            $curr = $curr->next;
            $count++;
        }
        
        $count /= 2;
        
        while ($count > 0)
        {
            $head = $head->next;
            $count--;
        }
        
        return $head;
    }
}

File: Coding Ninjas/Easy/Middle Of Linked List.java
public class Solution
{

    public static Node findMiddle
    ( Node head )
    {
        Node curr = head.next;

        while ( curr != null )
        {
            head = head.next;
            curr = curr.next;

            if ( curr != null )
            {
                curr = curr.next;
            }
        }

        return head;
    }

}
